### DISCUSSION TITLE
Add 'key' field to media folders

### DISCUSSION THREAD

**Discussion by sanderquirynen** [sulu/sulu#8033] [Q&A] [answered]
# Problem description

We've had a few occurrences where we needed to identify a specific folder through code, eg:
- A folder where users can drop images which will automatically be linked to a Product, based on the file name
- A folder with icons/illustrations we want to use for a certain single_media_selection field. We would then setup permissions so copywriters can select images, but not upload new ones. 

# Proposed solution

If Media folders had a 'key' field (like categories do), this would make it much easier to select a specific folder using PHP. This would also be cleaner than selecting by folder name or ID.
This would also make it relatively simple to add an extra parameter (e.g. '`folder_key`') to the `single_media_selection` and `media_selection` content types which would limit the CMS-user to only select files from a specific folder.

**Comment by alexander-schranz** [comment, marked as answer]
> If Media folders had a 'key' field (like categories do), this would make it much easier to select a specific folder using PHP. This would also be cleaner than selecting by folder name or ID.

A collection already has a `key` field which is for example used in the suluformbundle to store form media (also there is similar things implemented every page can have its own collection based on some [collection strategy](https://github.com/sulu/SuluFormBundle/blob/2.4/Resources/doc/dynamic.md)). They are because of the are programmatically only field not available for the content manager as it should not be possible to change that key. You can create this system collection the following way:

https://docs.sulu.io/en/2.4/cookbook/system-collections.html

And you can use the `$systemCollectionManager->getSystemCollection('my_key');` method then to get the `id` to store then the media then into that. In the form bundle the children are not configured system collection they the strategy create its own collection with an own `key` based on the page uuid. So you can do something similar in your case, that you configure the root collection as system collection and then create the other collection programmtically like we do it in the tree strategy of the sulu form bundle.
